crypto/pkcs12/p12_mutl.c

```diff
@@ -72,9 +94,66 @@ static int pkcs12_gen_gost_mac_key(const char *pass, int passlen,
     return 1;
 }
 
+static int pkcs12_gen_pbmac1_verify(OSSL_LIB_CTX *ctx, const char *propq,
+                                    const char *pass, int passlen,
+                                    const X509_ALGOR *macalg, unsigned char *key)
+{
+    PBMAC1PARAM *param = NULL;
+    PBKDF2PARAM *pbkdf2_param = NULL;
+    const ASN1_OBJECT *kdf_oid, *kdf_hmac_oid;
+    int ret = -1;
+    int keylen = 0;
+    EVP_MD *kdf_md = NULL;
+    const ASN1_OCTET_STRING *pbkdf2_salt = NULL;
+
+    param = ASN1_TYPE_unpack_sequence(ASN1_ITEM_rptr(PBMAC1PARAM), macalg->parameter);
+    if (param == NULL) {
+        ERR_raise(ERR_LIB_PKCS12, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
+
+    X509_ALGOR_get0(&kdf_oid, NULL, NULL, param->keyDerivationFunc);
+    if (OBJ_obj2nid(kdf_oid) != NID_id_pbkdf2) {
+        ERR_raise(ERR_LIB_PKCS12, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
+
+    pbkdf2_param = ASN1_TYPE_unpack_sequence(ASN1_ITEM_rptr(PBKDF2PARAM), param->keyDerivationFunc->parameter);
+    if (pbkdf2_param == NULL) {
+        ERR_raise(ERR_LIB_PKCS12, ERR_R_INTERNAL_ERROR);
+        goto err;
+    }
+    keylen = ASN1_INTEGER_get(pbkdf2_param->keylength);
```

Comment:
add the `RECOMMENDED` check to reject keylength smaller than 20 bytes from section 9?

Reply:
Not now